src/clients/c++/ensemble_image_client.cc

```diff
@@ -0,0 +1,263 @@
+namespace nic = nvidia::inferenceserver::client;
+
+#define FAIL_IF_ERR(X, MSG)                                        \
+  {                                                                \
+    nic::Error err = (X);                                          \
+    if (!err.IsOk()) {                                             \
+      std::cerr << "error: " << (MSG) << ": " << err << std::endl; \
+      exit(1);                                                     \
+    }                                                              \
+  }
+
+namespace {
+
+void
+Postprocess(
+    const std::map<std::string, std::unique_ptr<nic::InferContext::Result>>&
+        results,
+    const std::vector<std::string>& filenames, const size_t batch_size)
+{
+  if (results.size() != 1) {
+    std::cerr << "expected 1 result, got " << results.size() << std::endl;
+    exit(1);
+  }
+
+  const std::unique_ptr<nic::InferContext::Result>& result =
+      results.begin()->second;
+
+  for (size_t b = 0; b < batch_size; ++b) {
+    size_t cnt = 0;
+    nic::Error err = result->GetClassCount(b, &cnt);
+    if (!err.IsOk()) {
+      std::cerr << "failed reading class count for batch " << b << ": " << err
+                << std::endl;
+      exit(1);
+    }
+
+    std::cout << "Image '" << filenames[b] << "':" << std::endl;
+
+    for (size_t c = 0; c < cnt; ++c) {
+      nic::InferContext::Result::ClassResult cls;
+      nic::Error err = result->GetClassAtCursor(b, &cls);
+      if (!err.IsOk()) {
+        std::cerr << "failed reading class for batch " << b << ": " << err
+                  << std::endl;
+        exit(1);
+      }
+
+      std::cout << "    " << cls.idx << " (" << cls.label << ") = " << cls.value
+                << std::endl;
+    }
+  }
+}
+
+void
+Usage(char** argv, const std::string& msg = std::string())
+{
+  if (!msg.empty()) {
+    std::cerr << "error: " << msg << std::endl;
+  }
+
+  std::cerr << "Usage: " << argv[0] << " [options] <image filename / image folder>" << std::endl;
+  std::cerr << "\t-v" << std::endl;
+  std::cerr << "\t-i <Protocol used to communicate with inference service>"
+            << std::endl;
+  std::cerr << "\t-u <URL for inference service>" << std::endl;
+  std::cerr << std::endl;
+  std::cerr
+      << "For -i, available protocols are 'grpc' and 'http'. Default is 'http."
+      << std::endl;
+
+  exit(1);
+}
+
+}  // namespace
+
+int
+main(int argc, char** argv)
+{
+  bool verbose = false;
+  std::string url("localhost:8000");
+  std::string protocol = "http";
+
+  // Parse commandline...
+  int opt;
+  while ((opt = getopt(argc, argv, "vi:u:p:")) != -1) {
+    switch (opt) {
+      case 'v':
+        verbose = true;
+        break;
+      case 'i':
+        protocol = optarg;
+        break;
+      case 'u':
+        url = optarg;
+        break;
+      case '?':
+        Usage(argv);
+        break;
+    }
+  }
+
+  nic::Error err;
+
+  // The ensemble model takes 1 input tensor with shape [ 1 ] and STRING
```

Comment:
It would be good to add the -c flag so that the number of classification results can be selected.